src/transaction/log_replication_atomic.cpp

```diff
@@ -367,4 +370,26 @@ namespace cublog
 	  }
       }
   }
+
+  void
+  atomic_replicator::release_locks_by_tranid (cubthread::entry &thread_entry, const TRANID trid)
+  {
+    assert (m_locked_objects.count (trid) > 0);
+
+    for (auto it = m_locked_objects.lower_bound (trid); it != m_locked_objects.upper_bound (trid); it++)
+      {
+	lock_unlock_object (&thread_entry, & (it->second), oid_Root_class_oid, SCH_M_LOCK, true);
```

Comment:
Q: In the case of the Rootclass or non-mvcc class, is the last argument, `force`, is `true? Please tell me why shortly.

Reply:
OK, we should always make it call the `lock_internal_perform_unlock_object` as the `lock_unlock_all()` does when commit.

Another question: 
I assume we unlock the locks as it does when commit. In this regard, I see some differences and I'd like to know it's allowed.

Here: 
`lock_internal_perform_unlock_object (thread_p, entry_ptr, false, true);`
In the `lock_unlock_all()`, 
`lock_internal_perform_unlock_object (thread_p, entry_ptr, true, false);`

- in the `lock_unlock_all()`,
- /* remove non2pl locks */ part
- lock_clear_deadlock_victim

It could be a silly question. I just don't know. If you've already considered this, just answer it shortly please.

Reply:
Wow, thank you so much... I had overlooked that. 
If we only perform `lock_unlock_object`, the entries managed by the lock manager could have been left uncleared. Additionally, the non2pl entries would have kept accumulating. 
It seems like I should make a separate function to perform the unlock.

Reply:
Summary
1. Cleaning up the deadlock-related information seems a bit risky. The Replicator is registered with the lock manager as a single transaction, so it might affect other DDL statements. This issue appears to be something that needs to be addressed in LETS-787 (which deals with deadlock)
2. it may not be necessary to clean up the Non2pl list.
3. I will go ahead and implement the call to `lock_internal_perform_unlock_object(thread_p, entry_ptr, true, false)` as you suggested.

Detail
1. I wasn't sure if we could clear the deadlock in this PR.  
The Replicator should not become a victim when a deadlock occurs.  If we clear the deadlock information for the replicator, there might be issues with DDL transactions still in progress. 
 For example, 
  if in ATS, trx1 and trx2 are executing DDL simultaneously,  PTS can potentially trigger a deadlock while applying the changes made by trx1 and trx2.  
  Clearing deadlock information when trx1 commits may still affect trx2, which is still in progress,  because in PTS, the replicator is registered with the lock manager as a single transaction. 
  This part might need to be addressed in the lock manager extension (LETS-787).

2. The Non2pl list accumulates when a transaction releases a lock when it is not the end of the transaction. 
 As previously discussed, transactions sometimes temporarily use the lock manager to check the existence of a class and immediately release a lock, such as in xlocator_find_class_oid(). 
  These locks don't follow the 2PL rule, so they are stored in the Non2pl list and need to be cleaned up at commit time.   
  
  However, Non2pl entries are only created through `lock_unlock_object()`, and the replicator does not explicitly call `lock_unlock_object()`.  Therefore, there's no need to clean it up. When we use locator functions like xlocator_find_class_oid() in the future, we can clean up the Non2pl list as needed.
 
3. I will create a separate function to call `lock_internal_perform_unlock_object(thread_p, entry_ptr, true, false)` 
    - Since it's the commit or abort point, the third argument (release_flag) should be set to true, to clean up the lock entry. 
    - The replicator doesn't need to stack entries in this Non2pl list, so the last argument (move_to_non2pl) should be set to false.